homeassistant/components/osramlightify/light.py

```diff
@@ -4,270 +4,410 @@
-            for (group_name, group) in bridge.groups().items():
-                if group_name not in lights:
-                    osram_group = OsramLightifyGroup(
-                        group, bridge, update_lights)
-                    lights[group_name] = osram_group
-                    new_lights.append(osram_group)
+        return lights_changed
+
+    def update_groups():
+        """Update the groups objects with the latest info from the bridge."""
+        lights_changed = update_lights()
+
+        try:
+            new_groups = bridge.update_group_list(
+                config[CONF_INTERVAL_LIGHTIFY_CONF])
+            groups_updated = bridge.groups_updated()
+        except TimeoutError:
+            _LOGGER.error("Timeout during updating of groups")
+            return 0
+        except OSError:
+            _LOGGER.error("OSError during updating of groups")
+            return 0
+
+        if new_groups:
+            new_groups = {group.idx(): group for group in new_groups.values()}
+            new_entities = []
+            for idx, group in new_groups.items():
+                if idx not in groups:
+                    osram_group = OsramLightifyGroup(group, update_groups,
+                                                     groups_updated)
+                    groups[idx] = osram_group
+                    new_entities.append(osram_group)
                 else:
-                    lights[group_name].group = group
+                    groups[idx].update_luminary(group)
+
+            add_entities(new_entities)
+
+        if groups_updated > groups_last_updated[0]:
+            groups_last_updated[0] = groups_updated
+            for idx, osram_group in groups.items():
+                if idx not in new_groups:
+                    osram_group.update_static_attributes()
 
-        if new_lights:
-            add_entities(new_lights)
+        return max(lights_changed, groups_updated)
 
     update_lights()
+    if config[CONF_ALLOW_LIGHTIFY_GROUPS]:
+        update_groups()
 
 
 class Luminary(Light):
     """Representation of Luminary Lights and Groups."""
 
-    def __init__(self, luminary, update_lights):
-        """Initialize a Luminary light."""
-        self.update_lights = update_lights
+    def __init__(self, luminary, update_func, changed):
+        """Initialize a Luminary Light."""
+        self.update_func = update_func
         self._luminary = luminary
-        self._brightness = None
-        self._hs = None
-        self._name = None
-        self._temperature = None
-        self._state = False
-        self.update()
+        self._changed = changed
+
+        self._unique_id = None
+        self._supported_features = []
+        self._effect_list = []
+        self._is_on = False
+        self._min_mireds = 0
+        self._max_mireds = 0
+        self._brightness = 0
+        self._color_temp = 0
+        self._hs_color = (0, 0)
+        self._rgb_color = (0, 0, 0)
+        self._xy_color = (0, 0)
```

Comment:
Only support one color model. It looks like you should keep `_rgb_color` and delete `_hs_color` and `_xy_color`.

Reply:
Ok.  That's related to my comment above.

But it looks like I have to support `_hs_color` because parent `state_attributes` method has the following code:

```
if self.supported_features & SUPPORT_COLOR and self.hs_color:
```

There are no sections for `_rgb_color` or `_xy_color`.  Do I miss anything?

If needed, I can extend that method to support all three of them.  Please advise.

Reply:
You must support the `hs_color` property but you can do that while maintaining an internal `self._rgb_color` value. The internal value is up to you, I just thought RGB seemed to fit best in this case.

Reply:
Ok, got it!